cloud/previews/previews.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package previews
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/terramate-io/terramate/cloud"
+	"github.com/terramate-io/terramate/config"
+	"github.com/terramate-io/terramate/errors"
+)
+
+const (
+	// PreviewStatusAffected is the status for a stack that is affected in a PR
+	PreviewStatusAffected = "affected"
+	// PreviewStatusPending is the status for a stack that is selected in a preview run
+	PreviewStatusPending = "pending"
+)
+
+// RunContext is the context for a run
+type RunContext struct {
+	Stack *config.Stack
+	Cmd   []string
+}
```

Comment:
I don't get why this is leaking here. This type is a helper type of the cli package, also used by the other sync features.

Reply:
The preview package is unware of the `cli.runContext` type. The `preview.CreatePreview` function needs information about the all the scripts+commands that are part of a `--cloud-sync-preview` invocation, and this type + `[]RunContext` captures that information. It could have a better name though to reduce the confusion.